src/components/PopupWithForm.js

```diff
@@ -16,18 +17,24 @@ export default class PopupWithForm extends Popup {
   reset() {
     this._popupForm.reset();
   }
-  //{'#id1something':modal__input.value,'#id1something':modal__input.value}
 
   setInputValues(data) {
     this._inputEls.forEach((input) => {
       input.value = data[input.name];
     });
   }
+
   setEventListeners() {
     super.setEventListeners();
     this._popupForm.addEventListener("submit", (e) => {
       e.preventDefault();
       this._handleFormSubmit(this._getInputValues());
     });
   }
+  showUploadingState() {
+    this._submitBtn.innerText = "Saving";
+  }
+  showUploaded() {
+    this._submitBtn.innerText = "Saved";
```

Comment:
Needs correction: Here you should return the initial button text. For that you can save it in the constructor: `this._buttonText = this._submitBtn.textContent` and use `this._buttonText ` instead of `Saved` here